sysdeps/linux/generic/sysdeps.cpp

```diff
@@ -125,6 +125,45 @@ int sys_seek(int fd, off_t offset, int whence, off_t *new_offset) {
 	return 0;
 }
 
+int sys_chmod(const char *pathname, mode_t mode) {
+#ifdef SYS_chmod
+	auto ret = do_cp_syscall(SYS_chmod, pathname, mode);
+#else
+	auto ret = do_cp_syscall(SYS_fchmodat, AT_FDCWD, pathname, mode);
+#endif
```

Comment:
What's the advantage of this `#ifdef`? Can't we simply use SYS_fchmodat on all archs?

Reply:
Using the dedicated syscall, if available, feels like the correct choice to me. musl also prefers the dedicated syscall, if possible.

Reply:
We already use the more general syscalls where possible, so we should be consistent.  IMO less ifdefs make the code easier to follow and easier to test.